java.lang.RuntimeException: Unable to create application openfoodfacts.github.scrachx.openfood.views.OFFApplication: android.database.sqlite.SQLiteException: duplicate column name: quantity (code 1): , while compiling: ALTER TABLE history_product ADD COLUMN 'quantity' TEXT NOT NULL DEFAULT '';

03-03 14:08:24.772 19548-19548/openfoodfacts.github.scrachx.openfood E/SQLiteLog: (1) duplicate column name: quantity
03-03 14:08:24.791 19548-19548/openfoodfacts.github.scrachx.openfood E/AndroidRuntime: FATAL EXCEPTION: main
                                                                                       Process: openfoodfacts.github.scrachx.openfood, PID: 19548
                                                                                       java.lang.RuntimeException: Unable to create application openfoodfacts.github.scrachx.openfood.views.OFFApplication: android.database.sqlite.SQLiteException: duplicate column name: quantity (code 1): , while compiling: ALTER TABLE history_product ADD COLUMN 'quantity' TEXT NOT NULL DEFAULT '';
                                                                                           at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5743)
                                                                                           at android.app.ActivityThread.-wrap1(Unknown Source:0)
                                                                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656)
                                                                                           at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                           at android.os.Looper.loop(Looper.java:164)
                                                                                           at android.app.ActivityThread.main(ActivityThread.java:6494)
                                                                                           at java.lang.reflect.Method.invoke(Native Method)
                                                                                           at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
                                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
                                                                                        Caused by: android.database.sqlite.SQLiteException: duplicate column name: quantity (code 1): , while compiling: ALTER TABLE history_product ADD COLUMN 'quantity' TEXT NOT NULL DEFAULT '';
                                                                                           at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
                                                                                           at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:890)
                                                                                           at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:501)
                                                                                           at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:588)
                                                                                           at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:58)
                                                                                           at android.database.sqlite.SQLiteStatement.<init>(SQLiteStatement.java:31)
                                                                                           at android.database.sqlite.SQLiteDatabase.executeSql(SQLiteDatabase.java:1752)
                                                                                           at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java:1682)
                                                                                           at org.greenrobot.greendao.database.StandardDatabase.execSQL(StandardDatabase.java:37)
                                                                                           at openfoodfacts.github.scrachx.openfood.models.DatabaseHelper.upgrade(DatabaseHelper.java:53)
                                                                                           at openfoodfacts.github.scrachx.openfood.models.DatabaseHelper.onUpgrade(DatabaseHelper.java:30)
                                                                                           at org.greenrobot.greendao.database.DatabaseOpenHelper.onUpgrade(DatabaseOpenHelper.java:94)
                                                                                           at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:338)
                                                                                           at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:238)
                                                                                           at org.greenrobot.greendao.database.DatabaseOpenHelper.getWritableDb(DatabaseOpenHelper.java:59)
                                                                                           at openfoodfacts.github.scrachx.openfood.views.OFFApplication.onCreate(OFFApplication.java:48)
                                                                                           at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1119)
                                                                                           at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5740)
                                                                                           at android.app.ActivityThread.-wrap1(Unknown Source:0) 
                                                                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656) 
                                                                                           at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                           at android.os.Looper.loop(Looper.java:164) 
                                                                                           at android.app.ActivityThread.main(ActivityThread.java:6494) 
                                                                                           at java.lang.reflect.Method.invoke(Native Method) 
                                                                                           at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
                                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/models/DatabaseHelper.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/models/DatabaseHelper.java
@@ -37,7 +37,7 @@ public class DatabaseHelper extends DaoMaster.OpenHelper {
      * left untouched just fix the code in the version case and push a new
      * release
      *
-     * @param db database
+     * @param db             database
      * @param migrateVersion
      */
     private void upgrade(Database db, int migrateVersion) {
@@ -47,11 +47,14 @@ public class DatabaseHelper extends DaoMaster.OpenHelper {
                 break;
             case 3:
                 ToUploadProductDao.createTable(db, true);
+                break;
             case 4:
                 TagDao.createTable(db, true);
+                break;
             case 5: {
                 db.execSQL("ALTER TABLE history_product ADD COLUMN 'quantity' TEXT NOT NULL DEFAULT '';");
                 db.execSQL("ALTER TABLE history_product ADD COLUMN 'nutrition_grade' TEXT NOT NULL DEFAULT '';");
+                break;
             }
         }
     }

